v0.21: searchable attributes (#1055)
meilisearch/documentation: reference/api/searchable_attributes.md

Update the [searchable attributes](/reference/features/field_properties.md#searchable-fields) of an index.

::: warning

> Manually updating `searchableAttributes` will change the order of the returned documents' fields.

Reviewer: ```suggestion
Manually updating `searchableAttributes` will change the displayed order of the returned documents' fields in the JSON response.
```

Reviewer: Maybe my sentence is too long, I just don't want the users to think it will change another kind of order, like the attribute order for the relevancy

Author: This is great, just a minor alteration
```suggestion
Manually updating `searchableAttributes` will change the displayed order of document fields in the JSON response.
```

> Manually updating `searchableAttributes` will change the displayed order of document fields in the JSON response.

Reviewer 2: For me, it's not clear in which manner the order will be affected. Attributes are going to be displayed in the order specified in the `searchableAttributes` list?

Author: I think you are correct, but I'm not sure whether that's information we want to emphasise because this is a bug (or at least an unintended side-effect of other changes) which we will fix. Eventually. Meaning, even if it always follows the `searchableAttributes` order, I'm not sure we want people to rely on it or expect this will always be the case.

What do you think?

Reviewer 2: Well, even if it's a bug and it will change later, as a user I want to know what to expect if I update them manually... When we release another version we will update the information accordingly

Author: ```suggestion
Due to an implementation bug, manually updating `searchableAttributes` will change the displayed order of document fields in the JSON response. This behavior wields inconsistent results and will be fixed in a future release.
```

Reviewer 2: Oh, I didn't mean saying explicitly it is a bug, but explaining how is the order affected.  I was thinking more of something like: `Manually updating `searchableAttributes` will make the document fields in the JSON response appear in the same order of the searchableAttributes' list.` It's a horribly written sentence, but I am not inspired ATM 😅

Author: After speaking to @curquiza about this, although (from a quick test, at least) it seems like that is the case, we don't fully understand the mechanism behind this behaviour at the moment (or how to fix it). 

Since we can't describe it with 100% certainty, my goal here is to put a few red signs. Like, "this is a bug and you shouldn't rely on it".
